src/game.cpp

```diff
@@ -4458,6 +4459,33 @@ void Game::loadAccountStorageValues()
 	}
 }
 
+bool Game::loadItemsPrice()
+{
+	itemsSaleCount = 0;
+	std::ostringstream query, query2;
+	query << "SELECT DISTINCT `itemtype` FROM `market_offers`;";
+
+	Database& db = Database::getInstance();
+	DBResult_ptr result = db.storeQuery(query.str());
+	if (!result) {
+		return false;
+	}
+
+	do {
+		query2.str(std::string());
+		uint16_t itemId = result->getNumber<uint16_t>("itemtype");
+		query2 << "SELECT AVG(`price`) AS 'price' FROM `market_offers` WHERE `itemtype` = " << itemId << ";";
```

Comment:
This can be done in a single query using group by, instead of N+1 where N is itemtypes count.